app/controllers/api_keys_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ApiKeysController < ApplicationController
+  before_action :redirect_to_signin, unless: :signed_in?
+  before_action :find_api_key, only: :destroy
+
+  def index
+    @api_keys = current_user.api_keys
+    redirect_to new_profile_api_key_path if @api_keys.empty?
+  end
+
+  def new
+    @api_key = current_user.api_keys.build
+  end
+
+  def create
+    @key = SecureRandom.hex(16)
```

Comment:
Hi @sonalkr132, what do you think about making the new API keys a (more) regexable format, say with a common prefix? E.g. `<RUBYGEMS_SPECIFIC_PREFIX>_6b374974e76ad2849d894210d1bcfc09`. 

I suggest this as I suspect this would make integrating with GitHub's token scanning service a little easier in the future.

- https://developer.github.com/partnerships/token-scanning/#identify-your-tokens-and-create-regular-expressions
- https://help.github.com/en/articles/about-token-scanning